### PR TITLE
Imagely node v8

### PR DESCRIPTION
**Pull request by choshun**
Updated imagely to be compatible with node v8. This version uses [puppeteer](https://github.com/GoogleChrome/puppeteer) in place of phantom. Open to ideas, but it seems to work great so far.

All examples work, and from a users perspective should be completely the same, but faster (hopefully). 

The one caveat is that it is not backwards compatible with lower versions of node.

Further reading
* [Hopefully a meaningful benchmark](https://hackernoon.com/benchmark-headless-chrome-vs-phantomjs-e7f44c6956c)
* [Puppeteer API](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md)

### CODE REVIEW

**Comment by mpetrovich** [comment]
This is awesome! Puppeteer is great.

A couple things:
- Did you run it with the examples? (see the README) We should have automated tests for those, but for now manual tests will have to suffice.
- We use npm instead of yarn for this repo because [yarn doesn't work as needed when running in project subdirectories](https://github.com/yarnpkg/yarn/issues/2459) (NBS-PHP-Libs is what has this as a package, and NBS-PHP-Libs is installed as a subdirectory within NBS-Web and NBS-PHP-API during composer installation). Yet another reason to consider going back to npm across the board. For this PR, we need to update the `package-lock.json` and remove the `yarn.lock`.
- We'll need to add something like `"engines": { "node": ">=8" }` to `package.json` to make sure that this release will only get installed in compatible environments. See the [package.json "engines" docs](https://docs.npmjs.com/files/package.json#engines).
- We'll need to publish a new (patch) release after this is merged. I'll show you how.

**Comment by cynthiameng** [comment]
Death to Phantom. This is awesome!

**Review by cynthiameng — CHANGES_REQUESTED** [review]
Use npm and not yarn

**Comment by choshun** [comment]
@mpetrovich I ran all examples, even [fixed some bugs](https://github.com/nextbigsoundinc/imagely/pull/11/commits/f8e826a69707069fb31e94babccf056972204d5a) that were there before the update. 

I'll use npm, didn't know that , thanks!

Totally fine ditching yarn, especially after the new node.

**Review by cynthiameng — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -30,6 +30,9 @@
     "type": "git",
     "url": "git://github.com/nextbigsoundinc/imagely.git"
   },
+  "engines": {
+    "node": ">=6.4.0"
```
Comment (mpetrovich) [inline package.json:34]: I believe async/await is only available in Node 7.10.x+: https://node.green/#ES2017-features-async-functions

Comment (choshun) [reply package.json:34]: Awesome, thank you. Was looking at the min req for puppeteer

Will update

**Comment by mpetrovich** [comment]
I just remembered that we run Babel on prepublish (see package.json), so this should work on Node 4+ after all.

**Comment by mpetrovich** [comment]
Also, by using puppeteer we can remove a lot of the complexity in imagely. Things like the inlining of external js & css was done because phantomjs doesn’t support external js & css, but that works with puppeteer.

File: package.json (hunk 1 above)
Comment (mpetrovich) [reply package.json:34]: Nevermind, we're using Babel to transpile so this is moot

**Pull request merged**